Source/core/CyclicBuffer.cpp

```diff
@@ -243,7 +287,11 @@ namespace Core {
     void CyclicBuffer::AdminUnlock()
     {
 #ifdef __POSIX__
+#ifndef __DEBUG__
         pthread_mutex_unlock(&(_administration->_mutex));
+#else
```

Comment:
See above, ASSERTS might be useful/configured in REALEASE as well. Just allways ASSERT if you think it is usefull regardless of the BUILD type..